pkg/chaosdaemon/blockchaos_server.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2022 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package chaosdaemon
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	"github.com/chaos-mesh/chaos-driver/pkg/client"
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/pkg/errors"
+
+	"github.com/chaos-mesh/chaos-mesh/pkg/bpm"
+	pb "github.com/chaos-mesh/chaos-mesh/pkg/chaosdaemon/pb"
+)
+
+const chaosDaemonHelperCommand = "cdh"
+
+func (s *DaemonServer) ApplyBlockChaos(ctx context.Context, req *pb.ApplyBlockChaosRequest) (*pb.ApplyBlockChaosResponse, error) {
+	log := s.getLoggerFromContext(ctx)
+
+	volumeName, err := normalizeVolumeName(ctx, req.VolumePath)
```

Comment:
I think you can access the mnt ns of process in /proc/$PID/root/.

Reply:
@Andrewmatilde I have considered this problem, the difficult part is the symbolic path resolving. If there is a symlink points to an absolute address, e.g. `/home/vagrant/test -> /mnt`, then the `/proc/1/root/home/vagrant/test` in container also points to `/mnt`, which doesn't exist in the container. To resolve this problem we have to implement a symlink walking by ourselves, which is somehow disappointing, as it may contain relative paths, absolute paths...

Reply:
I agree with it.